dashboard/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -405,16 +405,21 @@ def silent_takeover(oauth_user, auth_hash)
           oauth_refresh_token: auth_hash.credentials&.refresh_token
         }.to_json
       )
-      unless success
+      unless option.persisted?
```

Comment:
This logic was actually broken before - `create` doesn't return boolean, it returns a model instance that may or may not be persisted.

Reply:
why not just `create!` above instead?